### ISSUE TITLE
[debug] cancellation of attach by PID leaves dangling promises

### ISSUE THREAD

**Issue by AlexTugarev** [eclipse-theia/theia#7316]
While testing https://github.com/eclipse-theia/theia/pull/7314 I learned about a dangling promise, which causing a never ending progress. When you try to attach to a running process, but then dismiss the quick open menu with `esc`, the awaiter will never continue at https://github.com/eclipse-theia/theia/blob/1ca3ec37c079a6b316278cf292609653e57b8151/packages/debug/src/browser/debug-session-manager.ts#L169

![2020-03-11 11 41 02](https://user-images.githubusercontent.com/914497/76412923-cd5daf00-6394-11ea-84f4-ae7781e19552.gif)

cc. @akosyakov

**Comment by akosyakov** [comment]
I think it is caused by https://github.com/eclipse-theia/theia/issues/5764 and https://github.com/eclipse-theia/theia/issues/5764.

The fix is here in gitlens PR: https://github.com/eclipse-theia/theia/pull/6921/commits/9c7d06474136aa36dbb0866b9e1451b63c80121e

I will extract it to a separate PR then.

**Comment by akosyakov** [comment]
No, it is something else. It looks like debugger can participate in the variable resolution: https://github.com/microsoft/vscode-node-debug/blob/501ceedca3b4e8aa1d0f62780558a0102dc8930a/package.json#L163-L165

So for us variable `PickProcess` never resolves to `extension.pickNodeProcess`

**Comment by akosyakov** [comment]
So it is combination of https://github.com/eclipse-theia/theia/issues/7316#issuecomment-597585425 and https://github.com/eclipse-theia/theia/issues/7316#issuecomment-598149095 what causing it.
